pkg/operator/missingstaticpodcontroller/missingstaticpodcontroller.go

```diff
@@ -0,0 +1,209 @@
+package missingstaticpodcontroller
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/informers"
+	corelisterv1 "k8s.io/client-go/listers/core/v1"
+)
+
+type MissingStaticPodController struct {
+	operatorClient v1helpers.StaticPodOperatorClient
+	podLister      corelisterv1.PodLister
+}
+
+// New checks the latest installer pods.  If the installer pod was successful, and if it has
+// been longer than the terminationGracePeriodSeconds+10seconds since the installer pod
+// completed successfully, and if the static pod is not at the correct revision, this
+// controller will go degraded.
+// It will also emit an event for detection in CI,
+func New(
+	operatorClient v1helpers.StaticPodOperatorClient,
+	kubeInformersForTargetNamespace informers.SharedInformerFactory,
+	eventRecorder events.Recorder,
+) factory.Controller {
+	c := &MissingStaticPodController{
+		operatorClient: operatorClient,
+		podLister:      kubeInformersForTargetNamespace.Core().V1().Pods().Lister(),
+	}
+	return factory.New().
+		WithInformers(
+			operatorClient.Informer(),
+			kubeInformersForTargetNamespace.Core().V1().Pods().Informer(),
+		).
+		WithSync(c.sync).
+		WithSyncDegradedOnError(operatorClient).
```

Comment:
I think it is ok to report a degraded condition whenever an error occurs since they are reported whenever we can't verify the status of a static pod.

As for reporting status degraded for missing static pods, we could accumulate all the missing pods into one error that would be reported inside of the degraded status, wdyt?

Reply:
> As for reporting status degraded for missing static pods, we could accumulate all the missing pods into one error that would be reported inside of the degraded status, wdyt?

yes, we should, otherwise, we might be updating the status too frequently.